packages/bot-skeleton/src/services/api/active-symbols.js

```diff
@@ -160,14 +186,97 @@ export default class ActiveSymbols {
                 if (submarket_name === submarket) {
                     const { symbols } = submarkets[submarket_name];
                     Object.keys(symbols).forEach(symbol_name => {
-                        accumulator.push([symbols[symbol_name].display_name, symbol_name]);
+                        const { display_name } = symbols[symbol_name];
+                        const symbol_display_name =
+                            display_name + (this.isSymbolClosed(symbol_name) ? ` ${localize('(Closed)')}` : '');
+                        accumulator.push([symbol_display_name, symbol_name]);
                     });
                 }
             });
 
             return accumulator;
         }, []);
 
-        return symbol_options.length === 0 ? config.NOT_AVAILABLE_DROPDOWN_OPTIONS : symbol_options;
+        if (symbol_options.length === 0) {
+            return config.NOT_AVAILABLE_DROPDOWN_OPTIONS;
+        }
+
+        return this.sortDropdownOptions(symbol_options, this.isSymbolClosed);
+    }
+
+    isMarketClosed(market_name) {
+        const market = this.processed_symbols[market_name];
+
+        if (!market) {
+            return true;
+        }
+
+        return Object.keys(market.submarkets).every(submarket_name => this.isSubmarketClosed(submarket_name));
+    }
+
+    isSubmarketClosed(submarket_name) {
+        const market_name = Object.keys(this.processed_symbols).find(name => {
+            const market = this.processed_symbols[name];
+            return Object.keys(market.submarkets).includes(submarket_name);
+        });
+
+        if (!market_name) {
+            return true;
+        }
+
+        const market = this.processed_symbols[market_name];
+        const submarket = market.submarkets[submarket_name];
+
+        if (!submarket) {
+            return true;
+        }
+
+        const { symbols } = submarket;
+        return Object.keys(symbols).every(symbol_name => this.isSymbolClosed(symbol_name));
     }
+
+    isSymbolClosed(symbol_name) {
+        let symbol = false;
+
+        Object.keys(this.processed_symbols).some(market_name => {
+            const market = this.processed_symbols[market_name];
+
+            return Object.keys(market.submarkets).some(submarket_name => {
+                const submarket = market.submarkets[submarket_name];
+
+                return Object.keys(submarket.symbols).some(curr_symbol_name => {
+                    if (symbol_name === curr_symbol_name) {
+                        symbol = submarket.symbols[curr_symbol_name];
+                        return true;
+                    }
+
+                    return false;
+                });
+            });
+        });
```

Comment:
you can simplify the code by using `this.active_symbols` to check whether symbol is active.
```
return this.active_symbols.includes(symbol => symbol.symbol === curr_symbol_name && (symbol.is_trading_suspended && !symbol.exchange_is_open));
```

Reply:
That makes much more sense 👍